File: server-scala-play-server/app/api/InviteApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.ConsumerInviteResponse
import model.InviteResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-11T19:31:43.310890579Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class InviteApiController @Inject()(cc: ControllerComponents, api: InviteApi) extends AbstractController(cc) {
  /**
    * POST /api/3.18/invite/accept?token=[value]&accountId=[value]&albumId=[value]&missionId=[value]&albumContestId=[value]&offerId=[value]&offerLocationId=[value]&retailerLocationId=[value]&appKey=[value]&autoFriend=[value]&autoAttendEvent=[value]&autoFavoriteOffer=[value]&autoFavoriteOfferLocation=[value]&autoFavoriteRetailerLocation=[value]
    */
  def acceptInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): ConsumerInviteResponse = {
      val token = request.getQueryString("token")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("token", "query string")
        }
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val albumId = request.getQueryString("albumId")
        .map(value => value.toLong)
        
      val missionId = request.getQueryString("missionId")
        .map(value => value.toLong)
        
      val albumContestId = request.getQueryString("albumContestId")
        .map(value => value.toLong)
        
      val offerId = request.getQueryString("offerId")
        .map(value => value.toLong)
        
      val offerLocationId = request.getQueryString("offerLocationId")
        .map(value => value.toLong)
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      val appKey = request.getQueryString("appKey")
        
      val autoFriend = request.getQueryString("autoFriend")
        .map(value => value.toBoolean)
        
      val autoAttendEvent = request.getQueryString("autoAttendEvent")
        .map(value => value.toBoolean)
        
      val autoFavoriteOffer = request.getQueryString("autoFavoriteOffer")
        .map(value => value.toBoolean)
        
      val autoFavoriteOfferLocation = request.getQueryString("autoFavoriteOfferLocation")
        .map(value => value.toBoolean)
        
      val autoFavoriteRetailerLocation = request.getQueryString("autoFavoriteRetailerLocation")
        .map(value => value.toBoolean)
        
      api.acceptInvite(token, accountId, albumId, missionId, albumContestId, offerId, offerLocationId, retailerLocationId, appKey, autoFriend, autoAttendEvent, autoFavoriteOffer, autoFavoriteOfferLocation, autoFavoriteRetailerLocation)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/invite/albumContest?deviceId=[value]&accountId=[value]&appId=[value]&appKey=[value]&albumContestId=[value]&latitude=[value]&longitude=[value]
    */
  def albumContestInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): InviteResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val appId = request.getQueryString("appId")
        .map(value => value.toLong)
        
      val appKey = request.getQueryString("appKey")
        
      val albumContestId = request.getQueryString("albumContestId")
        .map(value => value.toLong)
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      api.albumContestInvite(deviceId, accountId, appId, appKey, albumContestId, latitude, longitude)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/invite/album?deviceId=[value]&accountId=[value]&appId=[value]&appKey=[value]&albumId=[value]&latitude=[value]&longitude=[value]
    */
  def albumInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): InviteResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val appId = request.getQueryString("appId")
        .map(value => value.toLong)
        
      val appKey = request.getQueryString("appKey")
        
      val albumId = request.getQueryString("albumId")
        .map(value => value.toLong)
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      api.albumInvite(deviceId, accountId, appId, appKey, albumId, latitude, longitude)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/invite/event?accountId=[value]&receiverAccountIds=[value]&appKey=[value]&listingId=[value]&retailerLocationId=[value]
    */
  def eventInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): InviteResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val receiverAccountIds = request.getQueryString("receiverAccountIds")
        
      val appKey = request.getQueryString("appKey")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("appKey", "query string")
        }
        
      val listingId = request.getQueryString("listingId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("listingId", "query string")
        }
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      api.eventInvite(accountId, appKey, listingId, receiverAccountIds, retailerLocationId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/invite/gameLevel?deviceId=[value]&accountId=[value]&appId=[value]&appKey=[value]&gameLevelId=[value]&latitude=[value]&longitude=[value]
    */
  def gameInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): InviteResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val appId = request.getQueryString("appId")
        .map(value => value.toLong)
        
      val appKey = request.getQueryString("appKey")
        
      val gameLevelId = request.getQueryString("gameLevelId")
        .map(value => value.toLong)
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      api.gameInvite(deviceId, accountId, appId, appKey, gameLevelId, latitude, longitude)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/invite/get?accountId=[value]&token=[value]&albumId=[value]&missionId=[value]&albumContestId=[value]&offerId=[value]&offerLocationId=[value]&retailerLocationId=[value]&appKey=[value]
    */
  def getInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): SirqulResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val token = request.getQueryString("token")
        
      val albumId = request.getQueryString("albumId")
        .map(value => value.toLong)
        
      val missionId = request.getQueryString("missionId")
        .map(value => value.toLong)
        
      val albumContestId = request.getQueryString("albumContestId")
        .map(value => value.toLong)
        
      val offerId = request.getQueryString("offerId")
        .map(value => value.toLong)
        
      val offerLocationId = request.getQueryString("offerLocationId")
        .map(value => value.toLong)
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      val appKey = request.getQueryString("appKey")
        
      api.getInvite(accountId, token, albumId, missionId, albumContestId, offerId, offerLocationId, retailerLocationId, appKey)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/invite/mission?deviceId=[value]&accountId=[value]&appId=[value]&appKey=[value]&missionId=[value]&latitude=[value]&longitude=[value]
    */
  def missionInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): InviteResponse = {
      val deviceId = request.getQueryString("deviceId")
        
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        
      val appId = request.getQueryString("appId")
        .map(value => value.toLong)
        
      val appKey = request.getQueryString("appKey")
        
      val missionId = request.getQueryString("missionId")
        .map(value => value.toLong)
        
      val latitude = request.getQueryString("latitude")
        .map(value => value.toDouble)
        
      val longitude = request.getQueryString("longitude")
        .map(value => value.toDouble)
        
      api.missionInvite(deviceId, accountId, appId, appKey, missionId, latitude, longitude)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/invite/offer?accountId=[value]&appKey=[value]&offerId=[value]
    */
  def offerInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): InviteResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val appKey = request.getQueryString("appKey")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("appKey", "query string")
        }
        
      val offerId = request.getQueryString("offerId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("offerId", "query string")
        }
        
      api.offerInvite(accountId, appKey, offerId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/invite/offerLocation?accountId=[value]&appKey=[value]&offerLocationId=[value]
    */
  def offerLocationInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): InviteResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val appKey = request.getQueryString("appKey")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("appKey", "query string")
        }
        
      val offerLocationId = request.getQueryString("offerLocationId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("offerLocationId", "query string")
        }
        
      api.offerLocationInvite(accountId, appKey, offerLocationId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/invite/retailerLocation?accountId=[value]&appKey=[value]&retailerLocationId=[value]&albumId=[value]
    */
  def retailerLocationInvite(): Action[AnyContent] = Action { request =>
    def executeApi(): InviteResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val appKey = request.getQueryString("appKey")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("appKey", "query string")
        }
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("retailerLocationId", "query string")
        }
        
      val albumId = request.getQueryString("albumId")
        .map(value => value.toLong)
        
      api.retailerLocationInvite(accountId, appKey, retailerLocationId, albumId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}

File: server-java-play-framework/app/controllers/InviteApiController.java
package controllers;

import java.math.BigDecimal;
import apimodels.ConsumerInviteResponse;
import apimodels.InviteResponse;
import apimodels.SirqulResponse;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class InviteApiController extends Controller {
    private final InviteApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private InviteApiController(Config configuration, InviteApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result acceptInvite(Http.Request request, BigDecimal version) throws Exception {
        String valuetoken = request.getQueryString("token");
        String token;
        if (valuetoken != null) {
            token = valuetoken;
        } else {
            throw new IllegalArgumentException("'token' parameter is required");
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valuealbumId = request.getQueryString("albumId");
        Long albumId;
        if (valuealbumId != null) {
            albumId = Long.parseLong(valuealbumId);
        } else {
            albumId = null;
        }
        String valuemissionId = request.getQueryString("missionId");
        Long missionId;
        if (valuemissionId != null) {
            missionId = Long.parseLong(valuemissionId);
        } else {
            missionId = null;
        }
        String valuealbumContestId = request.getQueryString("albumContestId");
        Long albumContestId;
        if (valuealbumContestId != null) {
            albumContestId = Long.parseLong(valuealbumContestId);
        } else {
            albumContestId = null;
        }
        String valueofferId = request.getQueryString("offerId");
        Long offerId;
        if (valueofferId != null) {
            offerId = Long.parseLong(valueofferId);
        } else {
            offerId = null;
        }
        String valueofferLocationId = request.getQueryString("offerLocationId");
        Long offerLocationId;
        if (valueofferLocationId != null) {
            offerLocationId = Long.parseLong(valueofferLocationId);
        } else {
            offerLocationId = null;
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            appKey = null;
        }
        String valueautoFriend = request.getQueryString("autoFriend");
        Boolean autoFriend;
        if (valueautoFriend != null) {
            autoFriend = Boolean.valueOf(valueautoFriend);
        } else {
            autoFriend = true;
        }
        String valueautoAttendEvent = request.getQueryString("autoAttendEvent");
        Boolean autoAttendEvent;
        if (valueautoAttendEvent != null) {
            autoAttendEvent = Boolean.valueOf(valueautoAttendEvent);
        } else {
            autoAttendEvent = false;
        }
        String valueautoFavoriteOffer = request.getQueryString("autoFavoriteOffer");
        Boolean autoFavoriteOffer;
        if (valueautoFavoriteOffer != null) {
            autoFavoriteOffer = Boolean.valueOf(valueautoFavoriteOffer);
        } else {
            autoFavoriteOffer = false;
        }
        String valueautoFavoriteOfferLocation = request.getQueryString("autoFavoriteOfferLocation");
        Boolean autoFavoriteOfferLocation;
        if (valueautoFavoriteOfferLocation != null) {
            autoFavoriteOfferLocation = Boolean.valueOf(valueautoFavoriteOfferLocation);
        } else {
            autoFavoriteOfferLocation = false;
        }
        String valueautoFavoriteRetailerLocation = request.getQueryString("autoFavoriteRetailerLocation");
        Boolean autoFavoriteRetailerLocation;
        if (valueautoFavoriteRetailerLocation != null) {
            autoFavoriteRetailerLocation = Boolean.valueOf(valueautoFavoriteRetailerLocation);
        } else {
            autoFavoriteRetailerLocation = false;
        }
        return imp.acceptInviteHttp(request, version, token, accountId, albumId, missionId, albumContestId, offerId, offerLocationId, retailerLocationId, appKey, autoFriend, autoAttendEvent, autoFavoriteOffer, autoFavoriteOfferLocation, autoFavoriteRetailerLocation);
    }

    @ApiAction
    public Result albumContestInvite(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueappId = request.getQueryString("appId");
        Long appId;
        if (valueappId != null) {
            appId = Long.parseLong(valueappId);
        } else {
            appId = null;
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            appKey = null;
        }
        String valuealbumContestId = request.getQueryString("albumContestId");
        Long albumContestId;
        if (valuealbumContestId != null) {
            albumContestId = Long.parseLong(valuealbumContestId);
        } else {
            albumContestId = null;
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        return imp.albumContestInviteHttp(request, version, deviceId, accountId, appId, appKey, albumContestId, latitude, longitude);
    }

    @ApiAction
    public Result albumInvite(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueappId = request.getQueryString("appId");
        Long appId;
        if (valueappId != null) {
            appId = Long.parseLong(valueappId);
        } else {
            appId = null;
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            appKey = null;
        }
        String valuealbumId = request.getQueryString("albumId");
        Long albumId;
        if (valuealbumId != null) {
            albumId = Long.parseLong(valuealbumId);
        } else {
            albumId = null;
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        return imp.albumInviteHttp(request, version, deviceId, accountId, appId, appKey, albumId, latitude, longitude);
    }

    @ApiAction
    public Result eventInvite(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valuereceiverAccountIds = request.getQueryString("receiverAccountIds");
        String receiverAccountIds;
        if (valuereceiverAccountIds != null) {
            receiverAccountIds = valuereceiverAccountIds;
        } else {
            receiverAccountIds = null;
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            throw new IllegalArgumentException("'appKey' parameter is required");
        }
        String valuelistingId = request.getQueryString("listingId");
        Long listingId;
        if (valuelistingId != null) {
            listingId = Long.parseLong(valuelistingId);
        } else {
            throw new IllegalArgumentException("'listingId' parameter is required");
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        return imp.eventInviteHttp(request, version, accountId, appKey, listingId, receiverAccountIds, retailerLocationId);
    }

    @ApiAction
    public Result gameInvite(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueappId = request.getQueryString("appId");
        Long appId;
        if (valueappId != null) {
            appId = Long.parseLong(valueappId);
        } else {
            appId = null;
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            appKey = null;
        }
        String valuegameLevelId = request.getQueryString("gameLevelId");
        Long gameLevelId;
        if (valuegameLevelId != null) {
            gameLevelId = Long.parseLong(valuegameLevelId);
        } else {
            gameLevelId = null;
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        return imp.gameInviteHttp(request, version, deviceId, accountId, appId, appKey, gameLevelId, latitude, longitude);
    }

    @ApiAction
    public Result getInvite(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valuetoken = request.getQueryString("token");
        String token;
        if (valuetoken != null) {
            token = valuetoken;
        } else {
            token = null;
        }
        String valuealbumId = request.getQueryString("albumId");
        Long albumId;
        if (valuealbumId != null) {
            albumId = Long.parseLong(valuealbumId);
        } else {
            albumId = null;
        }
        String valuemissionId = request.getQueryString("missionId");
        Long missionId;
        if (valuemissionId != null) {
            missionId = Long.parseLong(valuemissionId);
        } else {
            missionId = null;
        }
        String valuealbumContestId = request.getQueryString("albumContestId");
        Long albumContestId;
        if (valuealbumContestId != null) {
            albumContestId = Long.parseLong(valuealbumContestId);
        } else {
            albumContestId = null;
        }
        String valueofferId = request.getQueryString("offerId");
        Long offerId;
        if (valueofferId != null) {
            offerId = Long.parseLong(valueofferId);
        } else {
            offerId = null;
        }
        String valueofferLocationId = request.getQueryString("offerLocationId");
        Long offerLocationId;
        if (valueofferLocationId != null) {
            offerLocationId = Long.parseLong(valueofferLocationId);
        } else {
            offerLocationId = null;
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            appKey = null;
        }
        return imp.getInviteHttp(request, version, accountId, token, albumId, missionId, albumContestId, offerId, offerLocationId, retailerLocationId, appKey);
    }

    @ApiAction
    public Result missionInvite(Http.Request request, BigDecimal version) throws Exception {
        String valuedeviceId = request.getQueryString("deviceId");
        String deviceId;
        if (valuedeviceId != null) {
            deviceId = valuedeviceId;
        } else {
            deviceId = null;
        }
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            accountId = null;
        }
        String valueappId = request.getQueryString("appId");
        Long appId;
        if (valueappId != null) {
            appId = Long.parseLong(valueappId);
        } else {
            appId = null;
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            appKey = null;
        }
        String valuemissionId = request.getQueryString("missionId");
        Long missionId;
        if (valuemissionId != null) {
            missionId = Long.parseLong(valuemissionId);
        } else {
            missionId = null;
        }
        String valuelatitude = request.getQueryString("latitude");
        Double latitude;
        if (valuelatitude != null) {
            latitude = Double.parseDouble(valuelatitude);
        } else {
            latitude = null;
        }
        String valuelongitude = request.getQueryString("longitude");
        Double longitude;
        if (valuelongitude != null) {
            longitude = Double.parseDouble(valuelongitude);
        } else {
            longitude = null;
        }
        return imp.missionInviteHttp(request, version, deviceId, accountId, appId, appKey, missionId, latitude, longitude);
    }

    @ApiAction
    public Result offerInvite(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            throw new IllegalArgumentException("'appKey' parameter is required");
        }
        String valueofferId = request.getQueryString("offerId");
        Long offerId;
        if (valueofferId != null) {
            offerId = Long.parseLong(valueofferId);
        } else {
            throw new IllegalArgumentException("'offerId' parameter is required");
        }
        return imp.offerInviteHttp(request, version, accountId, appKey, offerId);
    }

    @ApiAction
    public Result offerLocationInvite(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            throw new IllegalArgumentException("'appKey' parameter is required");
        }
        String valueofferLocationId = request.getQueryString("offerLocationId");
        Long offerLocationId;
        if (valueofferLocationId != null) {
            offerLocationId = Long.parseLong(valueofferLocationId);
        } else {
            throw new IllegalArgumentException("'offerLocationId' parameter is required");
        }
        return imp.offerLocationInviteHttp(request, version, accountId, appKey, offerLocationId);
    }

    @ApiAction
    public Result retailerLocationInvite(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueappKey = request.getQueryString("appKey");
        String appKey;
        if (valueappKey != null) {
            appKey = valueappKey;
        } else {
            throw new IllegalArgumentException("'appKey' parameter is required");
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            throw new IllegalArgumentException("'retailerLocationId' parameter is required");
        }
        String valuealbumId = request.getQueryString("albumId");
        Long albumId;
        if (valuealbumId != null) {
            albumId = Long.parseLong(valuealbumId);
        } else {
            albumId = null;
        }
        return imp.retailerLocationInviteHttp(request, version, accountId, appKey, retailerLocationId, albumId);
    }

}
